Modules/socketmodule.c

```diff
@@ -7668,6 +7668,42 @@ PyInit__socket(void)
     PyModule_AddIntConstant(m, "CAN_BCM_RX_STATUS", RX_STATUS);
     PyModule_AddIntConstant(m, "CAN_BCM_RX_TIMEOUT", RX_TIMEOUT);
     PyModule_AddIntConstant(m, "CAN_BCM_RX_CHANGED", RX_CHANGED);
+#ifdef  SETTIMER
```

Comment:
Why these separate `#ifdef` checks for each constant? Are they not either all available together, or not? More specifically, isn't the `#ifdef HAVE_LINUX_CAN_BCM_H` check enough to ensure their availability?

Reply:
This is not necessarily true for all the constants, namely the backwards-compatible introduction of the `CAN_FD_FRAME` flag, which was added in `6f3b911d5f29b98752e5da86a295210c0c4f4e14` as part of the the `4.8.x` kernel series.

The other constants were always present since the SocketCAN drivers were mainlined in `2.6.25`.

Should I just have the `#ifdef` on `CAN_FD_FRAME` then? I had added the other ones for consistency.

Reply:
> Should I just have the #ifdef on CAN_FD_FRAME then? I had added the other ones for consistency.

IMO that would be better, yes. And perhaps add a note about the availability of this flag in [the docs](https://docs.python.org/3/library/socket.html#socket.CAN_BCM).